src/Hyphenator/ArrayHyphenator.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Hyphenator;
+
+class ArrayHyphenator implements HyphenatorInterface
+{
+    private array $rules;
+
+    public function __construct(array $rules)
+    {
+        foreach($rules as $rule) {
+            $this->rules[] = new HyphenationRule($rule);
+        }
+    }
+
+    public function hyphenate(string $word): string
+    {
+        $word = new HyphenationWord($word);
+        foreach($this->rules as $rule) {
+            if (str_starts_with($rule->getRule(), '.') && $rule->matchesStart($word)) {
+                $iFrom = 0;
+                $iTo = min($word->getLevelsLength(), $rule->getLevelsLength());
```

Comment:
What does i stand for? 

![image](https://github.com/user-attachments/assets/1f4781fb-2195-4719-be42-546c20dd6a55)